### PR TITLE
eslint9 and flat config

### LINKED ISSUE #163: eslint v9 and flat config

### PR DESCRIPTION
**Pull request by le-cong**
Fixes #163

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,17 +19,19 @@ plugins that are supported by the configuration via `peerDependencies`.
...
     "lint": "eslint --max-warnings 0 --ignore-path .gitignore .",
     "lint:fix": "eslint --ignore-path .gitignore . --fix",
-    "test": "... && npm run lint"
-  }
-  "eslintConfig": {
-    "extends": [
-      "@checkdigit/eslint-config"
-    ]
-  }
+    "test": "... && npm run lint",
```
Comment (carlansley) [inline README.md:22]: we have to get rid of the --ignore-path stuff, correct?

File: README.md [hunk 2]
```diff
@@ -21,17 +19,19 @@ plugins that are supported by the configuration via `peerDependencies`.
...
+  },
   // ...
 }
 ```
 
+### Example `eslint.config.mjs`
+
+```javascript
+import config from '@checkdigit/eslint-config';
```
Comment (carlansley) [inline README.md:31]: `export { default } from '@checkdigit/eslint-config'`

File: index.mjs [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+import n from 'eslint-plugin-n';
+import tsParser from '@typescript-eslint/parser';
+import jest from 'eslint-plugin-jest';
+import js from '@eslint/js';
+import prettier from 'eslint-config-prettier';
+import { FlatCompat } from '@eslint/eslintrc';
+import unicorn from 'eslint-plugin-unicorn';
+
+const ignores = [
+  ...(await fs.readFile('.gitignore', 'utf-8')).split('\n').filter((path) => path.trim() !== ''),
```
Comment (carlansley) [inline index.mjs:27]: are we sure this is reading the files from the correct place?  i.e. from the project root, not from eslint-config root

**Review by carlansley — CHANGES_REQUESTED** [review]
obviously still needs the published version of eslint-plugin, but I have a few other minor things

File: index.mjs (hunk 3 above)
Comment (le-cong) [reply index.mjs:27]: yes, tested and confirmed that it's reading the file belonging to the current project instead of eslint-config module.

**Review by carlansley — APPROVED** [review]

**Pull request merged**